### PR TITLE
Discard alphabet when adding Seq objects

### PR DESCRIPTION
**Pull request by peterjc**
This pull request addresses in part issue #2046, and recent discussion on #3151 with the ``DBSeq`` which subclasses ``Seq``

This discards alphabets when adding Seq objects, plus a few other remaining cases, and removes alphabet usage within the ``Bio.Seq`` doctests.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
A little surprised that none of the tests failed - we must have already removed most of the alphabet checks.

**Comment by peterjc** [comment]
Must have got interrupted, currently testing another commit removing alphabet usage from more ``Bio/Seq.py`` doctests. Was sure I did this already today...

File: Bio/Seq.py [hunk 1]
```diff
@@ -1169,16 +1116,16 @@ class UnknownSeq(Seq):
     >>> print(unk_five)
     ?????
 
-    You can add unknown sequence together, provided their alphabets and
-    characters are compatible, and get another memory saving UnknownSeq:
+    You can add unknown sequence together. Provided the characters
+    are compatible, and get another memory saving UnknownSeq:
```
Comment (mdehoon) [inline Bio/Seq.py:1120]: grammar off

File: Bio/Seq.py [hunk 2]
```diff
@@ -1169,16 +1116,16 @@ class UnknownSeq(Seq):
...
 
     >>> unk_four = UnknownSeq(4)
     >>> unk_four
     UnknownSeq(4, character='?')
     >>> unk_four + unk_five
     UnknownSeq(9, character='?')
 
-    If the alphabet or characters don't match up, the addition gives an
+    If the characters don't match up, the addition gives an
```
Comment (mdehoon) [inline Bio/Seq.py:1128]: "match up" --> "are different"

**Comment by peterjc** [comment]
Thank you. Addressed.

File: Bio/Seq.py [hunk 3]
```diff
@@ -78,13 +79,13 @@ class Seq:
...
-    def __init__(self, data, alphabet=Alphabet.generic_alphabet):
+    def __init__(self, data, alphabet=None):
         """Create a Seq object.
 
         Arguments:
          - seq - Sequence, required (string)
-         - alphabet - Optional argument, an Alphabet object from
-           Bio.Alphabet
+         - alphabet - Optional argument, deprecated. Takes an Alphabet
+           object from Bio.Alphabet, default None.
```
Comment (mdehoon) [inline Bio/Seq.py:88]: "deprecated" is not quite the correct term here. "deprecated" suggests that the function for now still treats alphabets in the same way as before. But in reality we ignore the alphabet altogether. Also I am not sure where an alphabet argument could be coming from, as there won't be an `Alphabet` module any more. So if `alphabet` is not `None`, surely something is wrong, and we should raise a `ValueError`. In any case a `ValueError` is more user-friendly, as the traceback tells you explicitly where the problem is happening, while a `DeprecationWarning` just complains and then gives you the silent treatment.

Comment (peterjc) [reply Bio/Seq.py:88]: I am giving serious thought to leaving ``Bio.Alphabet`` present (but unused anywhere in Biopython) with a deprecation warning for two releases, so that historic code can continue to do things like ``Seq("aCGT", generic_dna)`` and it will run without changes (just with warnings pushing them to just ``Seq("aCGT")`` instead).

And for those cases where the alphabet was needed for an output format, add a few explicit lines like ``record.annotations["molecule_type"] = "DNA"`` to the code, and it should run on both recent Biopython and Biopython 1.78.

(This won't cover everything, but it would smooth the transition for many users)

Immediate removal forces people to make those code changes immediately if they want to use the latest Biopython.

Comment (mdehoon) [reply Bio/Seq.py:88]: Can you consider the advantages and disadvantages for the user of what you are proposing?

Comment (peterjc) [reply Bio/Seq.py:88]: That's what I'm trying to do. We might be best writing this down (as a separate issue on GitHub?), starting with some examples that could go into the deprecation file and/or extra chapter in the tutorial. Showing the old style, new style, and a hybrid which could be used if you needed to support both in the short term.

Comment (mdehoon) [reply Bio/Seq.py]: Well actually I was implying that the proposal to use a `DeprecatingWarning` has more disadvantages than advantages, and therefore we should not go down that route.

Comment (peterjc) [reply Bio/Seq.py]: Adding a deprecation warning may make writing backward compatible code easier, but it would be ugly - concrete example written up at #3156.

Retaining the stub alphabet argument (even if it must be ``None``) here in the ``Seq`` object and other widely used functions like ``Bio.SeqIO.read`` and ``Bio.SeqIO.parse`` is probably enough. Again, see example.

Comment (peterjc) [reply Bio/Seq.py]: I'll remove the deprecation warning here (and the other classes), leaving a TODO for a ValueError if not None.

**Comment by peterjc** [comment]
Ha - I see now that 1a60731b8d11cddae67d1f80b5e543dd0a48c0a4 was pushed to the branch, but that was meant as a local experiment. Oops. But worth having this discussion anyway.

File: Bio/Seq.py (hunk 3 above)
Comment (peterjc) [reply Bio/Seq.py]: (I see now that I accidentally included a test commit about the ``__init__`` arguments unrelated to the sequence addition work; removing that commit)

File: Bio/Seq.py [hunk 4]
```diff
@@ -75,7 +75,7 @@ class Seq:
 
     The Seq object also provides some biological methods, such as complement,
     reverse_complement, transcribe, back_transcribe and translate (which are
-    not applicable to sequences with a protein alphabet).
+    not applicable to protein sequences).
     """
 
     def __init__(self, data, alphabet=Alphabet.generic_alphabet):
```
Comment (mdehoon) [inline Bio/Seq.py:81]: The alphabet here will be removed in a separate PR?

Comment (peterjc) [reply Bio/Seq.py:81]: Yes the (removed) deprecation warning version showed we still use it but we're getting close.
